src/tty.rs

```diff
@@ -186,9 +186,20 @@ pub fn new<T: ToWinsize>(config: &Config, size: T) -> Pty {
 
     let (master, slave) = openpty(win.ws_row as _, win.ws_col as _);
 
-    let shell = config.shell().unwrap_or(pw.shell);
+    let (program, args) = match config.shell() {
```

Comment:
I get a type mismatch here. If `shell()` returns a `&Shell` and the closure returns a `Shell`. If I simply write `|| &Shell::new(pw.shell)` it won't live long enough. Maybe I'm doing something wrong with the lifetimes though. I haven' really understood lifetimes yet.